x-pack/plugin/security/src/main/java/org/elasticsearch/xpack/security/profile/ProfileService.java

```diff
@@ -101,13 +106,32 @@ public ProfileService(Settings settings, Clock clock, Client client, SecurityInd
         this.threadPool = threadPool;
     }
 
-    public void getProfile(String uid, @Nullable Set<String> dataKeys, ActionListener<Profile> listener) {
+    public void getProfile(String uid, Set<String> dataKeys, ActionListener<Profile> listener) {
         getVersionedDocument(
             uid,
             listener.map(versionedDocument -> versionedDocument != null ? versionedDocument.toProfile(dataKeys) : null)
         );
     }
 
+    public void getProfileSubjects(Collection<String> uids, ActionListener<Tuple<Map<String, Subject>, List<String>>> listener) {
```

Comment:
> If we return Profile instead of Subject, we can replace Map<String, Subject> with List<Subject>.

We discussed that this option would require retrieving the `Subject` of a `Profile`, which is a less favorable option (`Subject` is internal-ish and `Profile` is external facing).

> It may still be cumbersome to have Tuple<List<Subject>, List<String>>. We probably want to encapsulate them into its own class which can also provide an opportunity to return "not-found" UIDs in addition to errored ones (which, practicially, are more likely to happen).

I agree, pushed 55d6d442a441094b .